Sources/KituraWebSocket/WebSocketConnection.swift

```diff
@@ -97,6 +97,11 @@ extension WebSocketConnection: ChannelInboundHandler {
     public func channelRead(ctx: ChannelHandlerContext, data: NIOAny) {
         let frame = self.unwrapInboundIn(data)
 
+        if frame.rsv1 == true || frame.rsv2 == true || frame.rsv3 == true {
+            connectionClosed(reason: .protocolError, description: "RSV must be 0, when no extension defining RSV meaning has been negotiated")
```

Comment:
@nethraravindran  I think it would be better to have the error message indicate which of the RSV bits is invalid?

Also, the error message could be identical to the sentence in the RFC: `RSVx,[RSVy,RSVz] must be 0 unless an extension is negotiated that defines meanings for non-zero values`.

Reply:
@nethraravindran Can we make sure no extension has been negotiated? May be `frame.extensionData == nil` ?